File: aipolabs/common/schemas/project.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from aipolabs.common.enums import Visibility
from aipolabs.common.schemas.agent import AgentPublic


class ProjectCreate(BaseModel):
    """Project can be created under a user or an organization."""

    name: str
    organization_id: UUID | None = Field(
        default=None,
        description="Organization ID if project is to be created under an organization",
    )


class ProjectPublic(BaseModel):
    id: UUID
    owner_id: UUID
    name: str
    visibility_access: Visibility
    daily_quota_used: int
    daily_quota_reset_at: datetime
    total_quota_used: int

    created_at: datetime
    updated_at: datetime

    agents: list[AgentPublic]

    model_config = ConfigDict(from_attributes=True)

File: src/lib/types/project.ts
export interface Project {
    id: string;
    owner_id: string;
    name: string;
    visibility_access: string;
    daily_quota_used: number;
    daily_quota_reset_at: string;
    total_quota_used: number;
    created_at: string;
    updated_at: string;
    agents: Agent[];
}

export interface Agent {
    id: string;
    project_id: string;
    name: string;
    description: string;
    excluded_apps: string[];
    excluded_functions: string[];
    created_at: string;
    updated_at: string;
    api_keys: APIKey[];
}

export interface APIKey {
    id: string;
    key: string;
    agent_id: string;
    status: string;
    created_at: string;
    updated_at: string;
}
